src/com/esotericsoftware/kryo/util/Util.java

```diff
@@ -26,6 +26,9 @@
 /** A few utility methods, mostly for private use.
  * @author Nathan Sweet <misc@n4te.com> */
 public class Util {
+	//See: https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size
+	static public final int maxSafeArraySize = Integer.MAX_VALUE - 8;

```

Comment:
@magro thanks for the review. Updated:
 * use constant naming
 * moved to package private in `io`

PTAL